### PR TITLE
[config] Bring back default paths if the file exists

### PR DESCRIPTION
**Pull request by davidiw**
The goal of this commit is to satisfy two expectations:
    - a default node config can call load without failing
    - a default node config can load a these set of files if they exist in
    their default placement without the value being set in a node.config

    To address that, during load execution config checks
    - the config value matches its default
    - the config path is empty
    - the default path points to an actual file

    If all these are true, then it will set the default path rather than
    using an empty path.

### CODE REVIEW

File: config/src/config/consensus_config.rs [hunk 1]
```diff
@@ -23,6 +23,9 @@ use std::{collections::HashMap, path::PathBuf, sync::Arc};
 
 type ConsensusKeyPair = KeyPair<Ed25519PrivateKey>;
 
+const CONSENSUS_KEYPAIR_DEFAULT: &str = "consensus.keys.toml";
+const CONSENSUS_PEERS_DEFAULT: &str = "consensus_peers.config.toml";
```
Comment (mimoo) [inline config/src/config/consensus_config.rs]: 👍

File: config/src/config/execution_config.rs [hunk 2]
```diff
@@ -12,6 +12,8 @@ use std::{
     sync::Arc,
 };
 
+const GENESIS_DEFAULT: &str = "genesis.blob";
```
Comment (mimoo) [inline config/src/config/execution_config.rs]: btw shouldn't all these default files be sorted in `test_config` / `template` / `genesis` folders?

File: config/src/config/mod.rs [hunk 3]
```diff
@@ -110,6 +110,25 @@ impl BaseConfig {
             file_path.clone()
         }
     }
+
+    /// Returns true if the default is set, the current file path is empty, and the default file
```
Comment (mimoo) [inline config/src/config/mod.rs]: this doesn't return a bool

**Comment by andll** [comment]
@bors-libra delegate+

I am a bit worried that this diff did not fix issue - I'll leave it to your discretion whether we should merge it

File: config/src/config/mod.rs [hunk 4]
```diff
@@ -110,6 +110,25 @@ impl BaseConfig {
...
+    pub fn test_and_set_full_path<T>(
+        &self,
+        config_value: &T,
+        default_value: &T,
+        config_path: &mut PathBuf,
+        default_path: &str,
+    ) where
+        T: PartialEq,
+    {
+        if *config_path != PathBuf::new() || config_value != default_value {
```
Comment (mimoo) [inline config/src/config/mod.rs]: just a suggestion, but if you could have `config_path` be `&str` the empty check would be cleaner.

File: config/src/config/mod.rs [hunk 5]
```diff
@@ -110,6 +110,25 @@ impl BaseConfig {
...
     }
+
+    /// Returns true if the default is set, the current file path is empty, and the default file
+    /// path points to an existing file.
+    pub fn test_and_set_full_path<T>(
+        &self,
+        config_value: &T,
+        default_value: &T,
+        config_path: &mut PathBuf,
+        default_path: &str,
```
Comment (mimoo) [inline config/src/config/mod.rs]: why do we test both config_path and config_value? Can't we just check that config_path is not set?

**Pull request closed without merging**